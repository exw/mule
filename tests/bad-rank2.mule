let rank2 : all b. (all a. a -> a) -> {} =
  fn rank1. rank1 {}
in
# This should fail, since (fn _. 4) is only an *instance* of the required
# (polymorphic) type:
rank2 (fn _. 4)
