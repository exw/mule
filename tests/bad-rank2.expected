
Desugared:

	(let
	 (rank2
	  (fix/let
	   (fn rank2
	    ((: (all b ((all a (a -> a)) -> (record (types ()) (values ())))))
	     (fn rank1 (rank1 {}))))))
	 (rank2 (fn $0 (Integer 4))))
Type error: permission error during graft
